src/main/webapp/courseLoader.js

```diff
@@ -2,126 +2,216 @@ google.charts.load("current", { packages: ["corechart"] });
+  const profPerceptionList = [
+    ["Professor Perception"],
+    /* dummyPerceptionRating */ [2],
+    /* dummyPerceptionRating */ [9],
+  ].concat(tempProfPerceptionList);
+  const profPerceptionData = new google.visualization.arrayToDataTable(
+    profPerceptionList
+  );
+  const profPerceptionOptions = {
+    title: "Perception of Professor Reviews",
+    legend: { position: "none" },
+    vAxis: { title: "Perception" },
+    hAxis: { title: "Comment Quantity" },
+    histogram: {
+      hideBucketItems: true,
+    },
+  };
+  const profPerceptionChart = new google.visualization.Histogram(
+    document.getElementById("prof-chart")
+  );
+  profPerceptionChart.draw(profPerceptionData, profPerceptionOptions);
+}
+
+async function postData(url, data = {}) {
+  // Default options are marked with *
+  const response = await fetch(url, {
+    method: "POST", // *GET, POST, PUT, DELETE, etc.
+    cache: "no-cache",
+    credentials: "same-origin", // include, *same-origin, omit
+    headers: {
+      "Content-Type": "application/json",
+      // 'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: JSON.stringify(data), // body data type must match "Content-Type" header
+  });
+  return response.json(); // parses JSON response into native JavaScript objects
+}
+
+function storeData() {
+  var data = {};
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const courseName = urlParams.get("course-name");
+  const term = urlParams.get("term");
+  const profName = urlParams.get("prof-name");
+  const units = urlParams.get("num-units");
+  const schoolName = urlParams.get("school-name");
+
+  data["course-name"] = courseName;
+  data["term"] = term;
+  data["prof-name"] = profName;
+  data["units"] = units;
+  data["school-name"] = schoolName;
+
+  const termInput = document.getElementById("term-input").value;
+  const profInput = document.getElementById("prof-input").value;
+  const ratingTerm = document.getElementById("rating-term").value;
+  const ratingProf = document.getElementById("rating-prof").value;
+  const hours = document.getElementById("hoursOfWork").value;
+  const diff = document.getElementById("difficulty").value;
+  document.getElementById("termForm").reset();
+
+  data["term-input"] = termInput;
+  data["prof-input"] = profInput;
+  data["rating-term"] = ratingTerm;
+  data["rating-prof"] = ratingProf;
+  data["hours"] = hours;
+  data["difficulty"] = diff;
+
+  const url = newURL(
+    schoolName,
+    courseName,
+    profName,
+    units,
+    term,
+    termInput,
+    profInput,
+    ratingTerm,
+    ratingProf,
+    hours,
+    diff
+  );
+
+  return [url, data];
 }
 
+function newURL(
+  schoolName,
+  courseName,
+  profName,
+  units,
+  term,
+  termInput,
+  profInput,
+  ratingTerm,
+  ratingProf,
+  hours,
+  difficulty
+) {
+  const url = new URL("/data", window.location.origin);
+  url.searchParams.set("course-name", courseName);
```

Comment:
If you do keep the data object as dash case, you could pass it in directly here.

Then do this:

```
for(let key in data) {
  url.searchParams.set(key, data[key]);
}
```
Assuming you want all the properties on data to be search params.

Reply:
I am changing it to dash case and I later want to add the user ID to data and do not want that to be on the URL.

Reply:
I am changing it from*. It will now be camel case.